tooling/fuzzer/src/strategies/int.rs

```diff
@@ -40,35 +45,45 @@ impl IntStrategy {
             3 => self.type_max() - offset,
             _ => unreachable!(),
         };
-        Ok(proptest::num::i128::BinarySearch::new(start))
+        Ok(BinarySearch::new(start))
     }
 
     fn generate_random_tree(&self, runner: &mut TestRunner) -> NewTree<Self> {
         let rng = runner.rng();
-
         let start: i128 = rng.gen_range(self.type_min()..=self.type_max());
-        Ok(proptest::num::i128::BinarySearch::new(start))
+        Ok(BinarySearch::new(start))
     }
 
+    /// Maximum allowed positive number.
     fn type_max(&self) -> i128 {
-        if self.bits < 128 {
-            (1i128 << (self.bits - 1)) - 1
+        let bits = self.type_max_bits();
+        if bits < 128 {
```

Comment:
Wouldn't this always be true now?

Reply:
Yes; I thought, what if someone changes it back, it would fail to produce the maximum shift (at least in the `uint` version). 

But you're right, it looks daft. I moved the restriction to 64 bits to where the strategy is created, and then here it can still generate the maximum range, it just will never be asked to do so.